### PR TITLE
Minimum node version check before creating data stream

### PR DESCRIPTION
**Pull request by danhermann**
Relates to #53100

### CODE REVIEW

**Comment by danhermann** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java [hunk 1]
```diff
@@ -121,6 +122,10 @@ public CreateDataStreamClusterStateUpdateRequest(String name,
     static ClusterState createDataStream(MetadataCreateIndexService metadataCreateIndexService,
                                          ClusterState currentState,
                                          CreateDataStreamClusterStateUpdateRequest request) throws Exception {
+        if (currentState.nodes().getMinNodeVersion().compareTo(Version.V_8_0_0) < 0) {
```
Comment (martijnvg) [inline server/src/main/java/org/elasticsearch/cluster/metadata/MetadataCreateDataStreamService.java:125]: I think check is a bit clearer: `currentState.nodes().getMinNodeVersion().before(Version.V_8_0_0)`

**Review by martijnvg — APPROVED** [review]
I would expect the mixed version qa tests to fail, but I then realized that these tests are muted if version is in the 7.x range. We should version unmute them after this change has been backported.

LTGM otherwise.

**Comment by danhermann** [comment]
@elasticmachine update branch

**Pull request merged**